Add MatchState to store real match and alliance color

tails618 opened the pull request:
This commit adds match state tracking to the RobotContainer class, which includes a boolean flag for whether it is a real match and another boolean flag for the alliance color. The Robot class now updates the match state based on the DriverStation information. Additionally, in the DriveSubsystem class, the target heading is set to either 0 or 180 degrees depending on the alliance color.

^^ ai generated lmao

tails618 committed:
feat: Add match state tracking and set target heading based on alliance color

This commit adds match state tracking to the RobotContainer class, which includes a boolean flag for whether it is a real match and another boolean flag for the alliance color. The Robot class now updates the match state based on the DriverStation information. Additionally, in the DriveSubsystem class, the target heading is set to either 0 or 180 degrees depending on the alliance color.

balghane commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -62,6 +63,19 @@ public void autonomousInit() {
+
+    boolean realMatch = DriverStation.getMatchTime() > 1.0;
+    m_robotContainer.matchState.realMatch = realMatch;
+    // boolean blue = DriverStation.getAlliance() == DriverStation.Alliance.Blue;
```
Remove commented out code

balghane commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -77,6 +91,19 @@ public void teleopInit() {
+
+    boolean realMatch = DriverStation.getMatchTime() > 1.0;
+    m_robotContainer.matchState.realMatch = realMatch;
+    // boolean blue = DriverStation.getAlliance() == DriverStation.Alliance.Blue;
```
Remove

balghane commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -77,6 +91,19 @@ public void teleopInit() {
+    boolean realMatch = DriverStation.getMatchTime() > 1.0;
+    m_robotContainer.matchState.realMatch = realMatch;
+    // boolean blue = DriverStation.getAlliance() == DriverStation.Alliance.Blue;
+    if (!DriverStation.getAlliance().isEmpty()) {
+      boolean blue;
+      if (!realMatch) {
+        blue = true;
+      } else {
+        blue = DriverStation.getAlliance().get() == DriverStation.Alliance.Blue;
+      }
```
This section of code looks repeated, can you split it out

balghane commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -32,6 +32,18 @@ public RobotContainer() {
     configureBindings();
   }
 
+  public class MatchState {
+    public boolean realMatch;
+    public boolean blue;
+
+    public MatchState(boolean realMatch, boolean blue) {
+      this.realMatch = realMatch;
+      this.blue = blue;
+    }
+  }
```
Comments on class and variables and whatnot

balghane commented on src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java:
```diff
@@ -500,8 +501,8 @@ public void offsetCurrentHeading(double offsetDegrees) {
 
-  public void setZeroTargetHeading() {
-    targetHeadingDegrees = 0.0;
+  public void setForwardTargetHeading() {
+    targetHeadingDegrees = matchState.blue ? 0 : 180;
```
Was there anything else we needed to change? Maybe a 180 deg rotation in one of the drive commands? Some of those comments might be off

tails618 replied:
So I didn't see anything that seems like it would need to be adjusted, but I probably missed something.

tails618 committed:
Refactor match state logic into a separate method

- Extracted the match state logic into a new private method `setRobotContainerMatchState()`
- Improved code readability and maintainability by removing duplicate code in `autonomousInit()` and `teleopInit()`
- Updated documentation for `RobotContainer.MatchState` class to provide clearer explanations of its variables

tails618 committed:
Update field-relative speed calculation in DriveSubsystem

The commit updates the calculation of field-relative speeds in the DriveSubsystem class. It now takes into account the match state to determine whether to use positive or negative xSpeed values. This change ensures accurate field-relative movement during matches.

tails618 committed:
It needs to flip y-speed as well as x-speed

tails618 committed:
More changes

tails618 committed:
Clean up if statements

tails618 committed:
Revert "More changes"

This reverts commit 0432d94a1bb857fb0fe88a5ed79055c782348822.

tails618 committed:
Revert "It needs to flip y-speed as well as x-speed"

This reverts commit 6fb7c4905ff6934fb85c96778b547fa105e87cac.

tails618 committed:
Revert "Update field-relative speed calculation in DriveSubsystem"

This reverts commit 8dd833d3d91bb9f384b0fdeb52b110bdb9d8abbd.

balghane commented:
/format

github-actions[bot] committed:
Formatting fixes
